From 03093cfcd25a1623cddf58279075c4760bf72983 Mon Sep 17 00:00:00 2001
From: Ivan Shapovalov <intelfx@intelfx.name>
Date: Sat, 20 Aug 2016 03:57:22 +0300
Subject: [PATCH] liblmdb: add a CMake buildsystem

CMake simplifies building LMDB (which is widely used outside of OpenLDAP)
with nonstandard toolchains like Android CrystaX NDK which only support
their custom buildsystem and CMake (via a toolchain configuration file).
---
 libraries/liblmdb/CMakeLists.txt | 49 ++++++++++++++++++++++++++++++++++++++++
 1 file changed, 49 insertions(+)
 create mode 100644 libraries/liblmdb/CMakeLists.txt

diff --git a/libraries/liblmdb/CMakeLists.txt b/libraries/liblmdb/CMakeLists.txt
new file mode 100644
index 0000000..4c35e83
--- /dev/null
+++ b/libraries/liblmdb/CMakeLists.txt
@@ -0,0 +1,49 @@
+project(LMDB)
+cmake_minimum_required(VERSION 2.8)
+
+#
+# Private CFLAGS for all targets
+#
+add_compile_options(-W -Wall -Wno-unused-parameter -Wbad-function-cast -Wuninitialized)
+
+#
+# The liblmdb library and its public CFLAGS/LDFLAGS (transitively added to dependent targets)
+#
+add_library(lmdb
+            mdb.c midl.c)
+target_compile_options(lmdb PUBLIC -pthread)
+target_link_libraries(lmdb PUBLIC -pthread)
+
+install(TARGETS lmdb EXPORT LMDB
+        DESTINATION lib
+        INCLUDES DESTINATION include)
+install(FILES lmdb.h
+        DESTINATION include)
+install(EXPORT LMDB
+        DESTINATION lib/cmake/LMDB
+        FILE LMDBConfig.cmake)
+
+#
+# The installable binaries
+#
+foreach(binary mdb_stat mdb_copy mdb_dump mdb_load)
+	add_executable(${binary} ${binary}.c)
+	target_link_libraries(${binary} lmdb)
+
+	install(TARGETS ${binary}
+	        DESTINATION bin)
+endforeach()
+
+#
+# The test binaries
+#
+foreach(binary mtest mtest2 mtest3 mtest4 mtest5)
+	add_executable(${binary} ${binary}.c)
+	target_link_libraries(${binary} lmdb)
+endforeach()
+
+# ...and the ones failing to build
+foreach(binary mtest6)
+	add_executable(${binary} EXCLUDE_FROM_ALL ${binary}.c)
+	target_link_libraries(${binary} lmdb)
+endforeach()
-- 
2.9.3

